mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -123,7 +128,21 @@ class KMeansModel private[ml] (
   @Since("2.0.0")
   override def transform(dataset: Dataset[_]): DataFrame = {
     transformSchema(dataset.schema, logging = true)
-    val predictUDF = udf((vector: Vector) => predict(vector))
+    // val predictUDF = udf((vector: Vector) => predict(vector))
+    val predictUDF = if (dataset.schema($(featuresCol)).dataType.equals(new VectorUDT)) {
+      udf((vector: Vector) => predict(vector))
+    }
+    else {
+      udf((vector: Seq[_]) => {
```

Comment:
scala style: remove unnecessary ```{``` at end of line (IntelliJ should warn you about this)